plaid/src/executor/mod.rs

```diff
@@ -316,8 +318,58 @@ fn execution_loop(
             Some(module) => module,
         };
 
+        // If ANY of the modules that operate on this log type cannot be executed in parallel, we need to check
```

Comment:
This is not something I had considered: having a single threaded rule in a message type makes all things processing that log single threaded.

That is going to result in larger than expected performance hits when a non-concurrent rule is added to a logging channel.